lib/src/module.dart

```diff
@@ -28,7 +29,7 @@ import 'package:w_module/src/lifecycle_module.dart';
 ///   notification of internal module state change.
 /// - The [components] class exposes react-dart compatible UI components that
 ///   can be used to render module data.
-abstract class Module extends LifecycleModule {
+abstract class Module extends LifecycleModule implements SimpleModule {
```

Comment:
Could we have `LifeCycleModule` extend `SimpleModule` - then we wouldn't need the `api`, `components`, and `events` overrides in this class.